src/hotspot/share/gc/z/zStat.cpp

```diff
@@ -1270,88 +1273,65 @@ size_t ZStatHeap::capacity_low() {
               _at_relocate_end.capacity);
 }
 
-size_t ZStatHeap::available(size_t used) {
+size_t ZStatHeap::free(size_t used) {
   return _at_initialize.max_capacity - used;
 }
 
-size_t ZStatHeap::reserve(size_t used) {
-  return MIN2(_at_initialize.max_reserve, available(used));
+size_t ZStatHeap::allocated(size_t used, size_t reclaimed) {
+  return (used + reclaimed) - _at_mark_start.used;
```

Comment:
I don't know why used + reclaimed is the amount allocated. Is this correct? Could you add a comment?

Reply:
We do "used_now - used_at_some_earlier_point" to see how much the used grew (i.e. now much was allocated). However, if we just do that, we will fail to take into account that we might also have reclaimed memory, which would have lowered "used_now". So, we let "used_now" be "used + reclaimed" to get the used we would have seen if we had not reclaimed any memory. Hope that makes sense. I'll add a comment.